middleware.ts

```diff
@@ -0,0 +1,17 @@
+import { NextRequest, NextResponse } from "next/server";
+import { getToken } from "./api/requests";
+
+export async function middleware(request: NextRequest) {
+  // @ts-ignore
+  let authToken = request.cookies.get("authToken")?.value;
+  if (!authToken) {
+    authToken = await getToken();
+  }
+
+  const response = NextResponse.next();
+
+  // @ts-ignore
+  response.cookies.set("authToken", authToken);
```

Comment:
It's better to make "authToken" into a string constant. Maybe add COOKIES_KEY enum inside common/types add add authToken there.